src/pdftoraw.c

```diff
@@ -0,0 +1,210 @@
+{
+    return poppler_document_get_n_pages (doc);
+}
+
+static void
+pdf_page_raw_get (PopplerDocument *doc,
+                 int page_nr)
+{
+    cairo_surface_t *s = NULL;
+    cairo_t *cr = NULL;
+    unsigned char *data = NULL;
+    g_autoptr (PopplerPage) page = NULL;
+    g_autoptr (GdkPixbuf) pix = NULL;
+    double width = 0;
+    double height = 0;
+    int w = 0;
+    int h = 0;
+
+    page = poppler_document_get_page (doc, page_nr);
+    if (page == NULL)
+    {
+       g_error("failure poppler_document_get_page");
+       return;
+    }
+    poppler_page_get_size (page, &width, &height);
+    width = DPI * width / PTS;
+    height = DPI * height / PTS;
+    w = (int)ceil(width);
+    h = (int)ceil(height);
+    s = cairo_image_surface_create (CAIRO_FORMAT_ARGB32, w, h);
+    if (s == NULL)
+    {
+       g_error ("failure cairo_image_surface_create");
+       return;
+    }
+    cr = cairo_create (s);
+    if (cr == NULL)
+    {
+       cairo_surface_destroy (s);
+       g_error ("failure cairo_create");
+       return;
+    }
+    cairo_scale (cr, DPI / PTS, DPI / PTS);
+    cairo_save (cr);
+    poppler_page_render_for_printing (page, cr);
+    cairo_restore (cr);
+    pix = gdk_pixbuf_get_from_surface (s, 0, 0, w, h);
+    cairo_surface_destroy (s);
+    data = gdk_pixbuf_get_pixels (pix);
+    fwrite (data, 1, (w * h * 4), stdout);
+    cairo_destroy (cr);
+}
+
+static int
+pdf_page_width_get (PopplerDocument *doc,
+                   int page_nr)
+{
+    g_autoptr (PopplerPage) page = NULL;
+    double width = 0;
+    int w = 0;
+
+    page = poppler_document_get_page (doc, page_nr);
+    if (page == NULL)
+       return -1;
+    poppler_page_get_size (page, &width, NULL);
+    width = DPI * width / PTS;
+    w = (int)ceil (width);
+    return w;
+}
+
+static int
+pdf_page_height_get (PopplerDocument *doc,
+                    int page_nr)
+{
+    g_autoptr (PopplerPage) page = NULL;
+    double height = 0;
+    int h = 0;
+
+    page = poppler_document_get_page (doc, page_nr);
+    if (page == NULL)
+       return -1;
+    poppler_page_get_size (page, NULL, &height);
+    height = DPI * height / PTS;
+    h = (int)ceil (height);
+    return h;
+}
+
+int
+main (int argc, char **argv)
+{
+    g_autoptr (PopplerDocument) doc = NULL;
+    GError *err = NULL;
+    g_autoptr (GFile) in = NULL;
+
+    GOptionContext *context;
+
+    context = g_option_context_new ("- PDF utility for portal backends");
+    g_option_context_add_main_entries (context, entries, NULL);
+    if (!g_option_context_parse (context, &argc, &argv, &err))
+    {
+      g_printerr ("%s: %s", g_get_application_name (), err->message);
+      g_printerr ("\n");
+      g_printerr ("Try \"%s --help\" for more information.",
+                  g_get_prgname ());
+      g_printerr ("\n");
+      exit (EXIT_FAILURE);
+      return 1;
+    }
+
+    if (filename == NULL)
+    {
+        g_printerr ("%s", g_option_context_get_help (context, TRUE, NULL));
```

Comment:
Leak.

At the top do:

```c
g_autofree char *help = NULL;
```

and assign help here and below.

Reply:
Fixed